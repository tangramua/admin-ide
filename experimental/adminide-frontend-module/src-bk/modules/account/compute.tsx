
import * as  React from 'react';
import { PageWithTabs } from '@adminide-stack/react-shared-components';
import Tokens from './components/Tokens';
import Referals from './components/Referals';
import Integration from './components/Integration';
import Collaborators from './components/Collaborators';
import SharedConfiguration from './components/SharedConfiguration';
import Auth0Module, { AUTH0_NAMESPACE, AUTH0_ROUTES } from '@adminide-stack/user-auth0-browser';
import SubscriptionModule, { SUBSCRIPTION_ROUTES } from '@adminide-stack/subscription-stripe-browser';
import { IRouteData, IMenuData, IMenuPosition } from '@common-stack/client-react';
import { IPageStore } from '../../interfaces';
import { getFilteredMenus, getFilteredRoutes, getFilteredTabs } from '../../utils';

// Temporary workaround to activate routes
// in future this will be automated
const { routeConfig } = Auth0Module as any;
const { routeConfig: subscriptionRouteConfig } = SubscriptionModule;


const authRoutes = routeConfig[0];
const login = authRoutes[AUTH0_ROUTES.LOGIN];
const logout = authRoutes[AUTH0_ROUTES.LOGOUT];
const callback = authRoutes[AUTH0_ROUTES.CALLBACK];
const profile = authRoutes[AUTH0_ROUTES.PROFILE];

const subscribeRoutes = subscriptionRouteConfig[0];
const billing = subscribeRoutes[SUBSCRIPTION_ROUTES.BILLING];
const payments = subscribeRoutes[SUBSCRIPTION_ROUTES.PAYMENT];
const paymentReports = subscribeRoutes[SUBSCRIPTION_ROUTES.REPORT];

const Account = (props) => (
    <PageWithTabs {...props} pageHeader={'Account Settings'} tabs={filteredTabs} />
);

const Billing = (props) => (
    <PageWithTabs {...props} pageHeader={'Billing'} tabs={filteredBillingTabs} />
);

export const accountPageStore: IPageStore[] = [
    {
        key: 'usermenu',
        tab: 'User Menu',
        path: '/usermenu',
        position: IMenuPosition.BOTTOM,
        exact: false,
    },
    {
        key: 'login',
        path: '/login',
        ...login,
    },
    {
        key: 'callback',
        path: '/callback',
        ...callback,
    },
    {
        key: 'billing',
        tab: 'Billing',
        path: '/usermenu/billing',
        exact: false,
        component: Billing,
    },
    {
        key: 'plans',
        tab: 'Plans',
        path: '/usermenu/billing/plans',
        ...billing,
    },
    {
        key: 'payments',
        tab: 'Payments',
        path: '/usermenu/billing/payments',
        ...payments,
    },
    {
        key: 'paymentReports',
        tab: 'Reports',
        path: '/usermenu/billing/reports',
        ...paymentReports,
    },
    {
        key: 'account',
        tab: 'Accounts',
        path: '/usermenu/account',
        exact: false,
        component: Account,
    },
    {
        ...logout,
        key: 'logout',
        name: 'Logout',
        path: '/usermenu/logout',
    },
    {
        key: 'profile',
        tab: 'Profile',
        name: 'Profile Settings',
        path: '/usermenu/account/profile',
        ...profile,
    },
    {
        key: 'collaborators',
        tab: 'Collaborators',
        path: '/usermenu/account/collaborators',
        component: Collaborators,
    },
    {
        key: 'integration',
        tab: 'Integration',
        path: '/usermenu/account/integration',
        component: Integration,
    },
    {
        key: 'tokens',
        tab: 'Tokens',
        path: '/usermenu/account/tokens',
        component: Tokens,
    },
    {
        key: 'referrals',
        tab: 'Referrals',
        path: '/usermenu/account/referrals',
        component: Referals,
    },
];

const selectedBillingTabs = ['plans', 'payments', 'paymentReports'];

const filteredBillingTabs = getFilteredTabs(accountPageStore, selectedBillingTabs);

// get only selected tabs to show
const selectedTabs = ['profile', 'collaborators', 'integration', 'tokens', 'referrals'];

const filteredTabs = getFilteredTabs(accountPageStore, selectedTabs);

// get only selected menus to show
const selectedMenu = ['billing', 'usermenu', 'logout', 'account', 'profile', 'collaborators', 'integration',
    'tokens', 'referrals', 'plans', 'payments', 'paymentReports'];

const filteredMenus: IMenuData[] = getFilteredMenus(accountPageStore, selectedMenu);

// get only selected Routes to work
const selectedRoutes = ['billing', 'login', 'logout', 'callback',
    'account', 'profile', 'collaborators', 'integration', 'tokens', 'referrals', 'plans', 'payments', 'paymentReports'];

const filteredRoutes: IRouteData[] = getFilteredRoutes(accountPageStore, selectedRoutes);

export { filteredRoutes, filteredMenus, filteredTabs, filteredBillingTabs };
