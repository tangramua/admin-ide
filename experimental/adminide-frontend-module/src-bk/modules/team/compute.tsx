import * as React from 'react';
import { IMenuPosition } from '@common-stack/client-react';
import { userIsAuthenticatedRedir } from '@adminide-stack/user-auth0-browser';

import { TeamPage, TeamForm, InvitationPage } from './containers';
import { getFilteredMenus, getFilteredRoutes, getFilteredTabs } from '../../utils';

export const team = [
    {
        path: '/teams/view/:team',
        key: 'team-management',
        component: userIsAuthenticatedRedir(TeamPage),
    },
    {
        name: 'Teams',
        path: '/teams/dashboard',
        icon: 'usergroup-add',
        key: 'team-creation',
        component: userIsAuthenticatedRedir(TeamForm),
    },
    {
        name: 'Team Invitation',
        key: 'team-invitation',
        path: '/invitation/:invitation',
        component: userIsAuthenticatedRedir(InvitationPage),
    },
];

const selectedMenus = ['team-management', 'team-creation'];
const selectedRoutes = ['team-management', 'team-creation', 'team-invitation'];

// get menus
const filteredMenus = getFilteredMenus(team, selectedMenus);

// get routes
const filteredRoutes = getFilteredRoutes(team, selectedRoutes);

export { filteredMenus, filteredRoutes };
