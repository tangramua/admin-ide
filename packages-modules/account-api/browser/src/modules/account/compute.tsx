
import * as  React from 'react';
import { PageWithTabs } from '@adminide-stack/react-shared-components';
import Tokens from './components/Tokens';
import Referals from './components/Referals';
import Integration from './components/Integration';
import Collaborators from './components/Collaborators';
import { IRouteData, IMenuData, IMenuPosition } from '@common-stack/client-react';
import { IPageStore, getFilteredMenus, getFilteredRoutes, getFilteredTabs  } from '@adminide-stack/core';





const Account = (props) => (
    <PageWithTabs {...props} pageHeader={'Account Settings'} tabs={filteredTabs} />
);

export const accountPageStore: IPageStore[] = [
    {
        key: 'usermenu',
        tab: 'User Menu',
        path: '/usermenu',
        position: IMenuPosition.BOTTOM,
        exact: false,
    },
    {
        key: 'account',
        tab: 'Accounts',
        path: '/usermenu/account',
        exact: false,
        component: Account,
    },
    {
        key: 'collaborators',
        tab: 'Collaborators',
        path: '/usermenu/account/collaborators',
        component: Collaborators,
    },
    {
        key: 'integration',
        tab: 'Integration',
        path: '/usermenu/account/integration',
        component: Integration,
    },
    {
        key: 'tokens',
        tab: 'Tokens',
        path: '/usermenu/account/tokens',
        component: Tokens,
    },
    {
        key: 'referrals',
        tab: 'Referrals',
        path: '/usermenu/account/referrals',
        component: Referals,
    },
];


// get only selected tabs to show
const selectedTabs = ['profile', 'collaborators', 'integration', 'tokens', 'referrals'];

const filteredTabs = getFilteredTabs(accountPageStore, selectedTabs);

// get only selected menus to show
const selectedMenu = ['usermenu', 'logout', 'account', 'profile', 'collaborators', 'integration',
    'tokens', 'referrals'];

const filteredMenus: IMenuData[] = getFilteredMenus(accountPageStore, selectedMenu);

// get only selected Routes to work
const selectedRoutes = ['login', 'logout', 'callback',
    'account', 'profile', 'collaborators', 'integration', 'tokens', 'referrals'];

const filteredRoutes: IRouteData[] = getFilteredRoutes(accountPageStore, selectedRoutes);

export { filteredRoutes, filteredMenus, filteredTabs };
