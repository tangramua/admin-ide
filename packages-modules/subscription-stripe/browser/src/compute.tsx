import * as React from 'react';
import Billing from './components/Billing';
import Payments from './components/Payments';
import Reports from './components/Reports';
import { PageWithTabs } from '@adminide-stack/react-shared-components';
import { IPageStore, getFilteredMenus, getFilteredRoutes, getFilteredTabs } from '@adminide-stack/core';
import { IRouteData, IMenuData, IMenuPosition } from '@common-stack/client-react';
import { SUBSCRIPTION_ROUTES } from './constants';


const defaultRouteConfig = [
    {
        [SUBSCRIPTION_ROUTES.BILLING]: { component: Billing },
        [SUBSCRIPTION_ROUTES.PAYMENT]: { component: Payments },
        [SUBSCRIPTION_ROUTES.REPORT]: { component: Reports },
    },
];


const BillingTab = (props) => (
    <PageWithTabs {...props} pageHeader={'Billing'} tabs={filteredBillingTabs} />
);

const billingPageStore: IPageStore[] = [
    // this would define the parent menu position of the billing menus
    {
        key: 'usermenu',
        tab: 'User Menu',
        path: '/usermenu',
        position: IMenuPosition.BOTTOM,
        exact: false,
    },
    // this would deine the submenu's parent of Billing
    {
        key: 'billing',
        tab: 'Billing',
        path: '/usermenu/billing',
        exact: false,
        component: BillingTab,
    },
    {
        key: 'plans',
        tab: 'Plans',
        path: '/usermenu/billing/plans',
        component: Billing,
    },
    {
        key: 'payments',
        tab: 'Payments',
        path: '/usermenu/billing/payments',
        component: Payments,
    },
    {
        key: 'paymentReports',
        tab: 'Reports',
        path: '/usermenu/billing/reports',
        component: Reports,
    },
];


const selectedBillingTabs = ['plans', 'payments', 'paymentReports'];
const filteredBillingTabs = getFilteredTabs(billingPageStore, selectedBillingTabs);


const selectedMenu = ['usermenu', 'billing', 'plans', 'payments', 'paymentReports'];
const filteredMenus: IMenuData[] = getFilteredMenus(billingPageStore, selectedMenu);


const selectedRoutes = ['billing', 'plans', 'payments', 'paymentReports'];
const filteredRoutes: IRouteData[] = getFilteredRoutes(billingPageStore, selectedRoutes);



export {
    filteredMenus,
    filteredBillingTabs,
    filteredRoutes,
    defaultRouteConfig,
};
