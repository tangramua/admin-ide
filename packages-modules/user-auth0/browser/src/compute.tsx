
import * as  React from 'react';

import { IPageStore, getFilteredMenus, getFilteredRoutes  } from '@adminide-stack/core';
import { IRouteData, IMenuData, IMenuPosition } from '@common-stack/client-react';
import { AUTH0_NAMESPACE, AUTH0_ROUTES } from './constants';
import Login from './components/Login';
import Callback from './components/Callback';
import Logout from './components/Logout';
import Profile from './components/Profile';



export const defaultRouteConfig  = [
    {
        [AUTH0_ROUTES.LOGIN]: {
          component: Login,
        },
        [AUTH0_ROUTES.CALLBACK]: {
          component: Callback,
        },
        [AUTH0_ROUTES.LOGOUT]: {
          component: Logout,
        },
        [AUTH0_ROUTES.PROFILE]: {
          component: Profile,
        },
      },
];

export const authPageStore: IPageStore[] = [
    {
        key: 'usermenu',
        tab: 'User Menu',
        path: '/usermenu',
        position: IMenuPosition.BOTTOM,
        exact: false,
    },
    {
        key: 'login',
        path: '/login',
        component: Login,
    },
    {
        key: 'callback',
        path: '/callback',
        component: Callback,
    },
    {
        key: 'logout',
        name: 'Logout',
        path: '/usermenu/logout',
        component: Logout,
    },
    {
        key: 'profile',
        tab: 'Profile',
        name: 'Profile Settings',
        path: '/usermenu/account/profile',
        component: Profile,
    },

];


// get only selected menus to show
const selectedMenu = ['usermenu', 'logout', 'account', 'profile' ];

const filteredMenus: IMenuData[] = getFilteredMenus(authPageStore, selectedMenu);

// get only selected Routes to work
const selectedRoutes = ['login', 'logout', 'callback',
    'account', 'profile'];

const filteredRoutes: IRouteData[] = getFilteredRoutes(authPageStore, selectedRoutes);


export { filteredRoutes, filteredMenus };

