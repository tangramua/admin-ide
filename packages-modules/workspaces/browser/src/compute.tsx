import * as React from 'react';
import { IMenuPosition } from '@common-stack/client-react';
import { userIsAuthenticatedRedir, isUserAuthorized } from '@adminide-stack/user-auth0-browser';
import { getFilteredMenus, getFilteredRoutes, getFilteredTabs, IPageStore } from '@adminide-stack/core';
import MonocularModule, { MONOCULAR_ROUTES } from '@adminide-stack/monocular-api-browser';
import { NewWorkspace as NewWorkspaceComponent } from './containers/NewWorkspace';
import { WorkspaceEditor as WorkspaceEditorComponent } from './containers/WorkspaceEditor';
import { Details as DetailsComponent } from './containers';
import { WorkspaceEditorIFrameComponent } from './components';
import { Dashboard } from './components/Dashboard';
import { Registries } from './containers/Registries';

const { routeConfig } = MonocularModule;
const moncoularRoutes = routeConfig[0];
const charts = moncoularRoutes[MONOCULAR_ROUTES.CHARTS];
const chart = moncoularRoutes[MONOCULAR_ROUTES.CHART];
const chartRepo = moncoularRoutes[MONOCULAR_ROUTES.REPO_CHART];

const NewWorkspace = (props) => <NewWorkspaceComponent {...props} />;
const Details = (props) => <DetailsComponent {...props} />;
const WorkspaceEditor = (props) => <WorkspaceEditorComponent {...props} />;


export const workspacePageStore: IPageStore[] = [
    {
        key: 'dashboard',
        path: '/dashboard',
        position: IMenuPosition.MIDDLE,
        name: 'Dashboard',
        exact: true,
        icon: 'appstore-o',
        component: isUserAuthorized(Dashboard),
    },
    {
        key: 'registries',
        path: '/registries',
        position: IMenuPosition.MIDDLE,
        name: 'Registries',
        exact: true,
        icon: 'api',
        component: isUserAuthorized(Registries),
    },
    {
        key: 'create-workspace',
        path: '/create-workspace',
        position: IMenuPosition.MIDDLE,
        name: 'Create Workspace',
        exact: true,
        icon: 'plus-circle-o',
        component: userIsAuthenticatedRedir(NewWorkspace),
    },
    {
        key: 'workspace-details',
        path: '/workspace/:id/details',
        position: IMenuPosition.MIDDLE,
        name: 'Workspace Details',
        exact: true,
        component: userIsAuthenticatedRedir(Details),
    },
    {
        key: 'workspace-editor',
        path: '/workspace/:id/editor',
        position: IMenuPosition.MIDDLE,
        name: 'Workspace Editor',
        exact: true,
        component: userIsAuthenticatedRedir(WorkspaceEditor),
    },
    {
        key: 'stack',
        path: '/stack',
        position: IMenuPosition.MIDDLE,
        name: 'Stack',
        ...chart,
    },
    {
        key: 'stack-charts',
        path: '/stack/charts',
        icon: 'compass',
        position: IMenuPosition.MIDDLE,
        name: 'Stack',
        ...charts,
        exact: true,
    },
    {
        key: 'stack-repo-chart',
        path: '/stack/:registry/:repo/:chart',
        position: IMenuPosition.MIDDLE,
        name: 'Stack Chart',
        ...chartRepo,
    },
    {
        key: 'workspace',
        path: '/workspace/:id',
        position: IMenuPosition.MIDDLE,
        name: 'Workspace',
        exact: true,
    },
];


// get routes
const selectedRoutes = ['dashboard', 'workspace', 'registries', 'stack-charts', 'stack', 'stack-repo-chart', 'workspace-details',
    'workspace-editor', 'create-workspace'];
const filteredRoutes = getFilteredRoutes(workspacePageStore, selectedRoutes);

// get menus
const selectedMenus = ['dashboard', 'create-workspace', 'stack-charts', 'registries'];
const filteredMenus = getFilteredMenus(workspacePageStore, selectedMenus);

export { filteredRoutes, filteredMenus };
