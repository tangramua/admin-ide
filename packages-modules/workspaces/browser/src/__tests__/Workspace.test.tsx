import * as React from 'react';

import { Feature } from '@common-stack/client-react';

import { filteredMenus, filteredRoutes } from '../compute';
import 'jest';


describe('Workspace Page', () => {

    it('testing the pageStore', () => {

        expect(filteredMenus).toMatchSnapshot();
        expect(filteredRoutes).toMatchSnapshot();

    });

    it('checking generatedRoutes', () => {
        const connector = new Feature({routeConfig: filteredRoutes});

        const configuredRoutes = connector.getConfiguredRoutes();
        expect(configuredRoutes).toMatchSnapshot();
        const routes = connector.getRoutes();

        expect(routes).toMatchSnapshot();

    });
});

