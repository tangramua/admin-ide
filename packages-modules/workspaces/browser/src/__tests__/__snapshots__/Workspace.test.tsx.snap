// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace Page checking generatedRoutes 1`] = `
Array [
  Object {
    "component": [Function],
    "exact": true,
    "path": "/create-workspace",
  },
  Object {
    "component": [Function],
    "exact": true,
    "path": "/workspace/:id/details",
  },
  Object {
    "component": [Function],
    "exact": true,
    "path": "/workspace/:id/editor",
  },
  Object {
    "component": [Function],
    "exact": false,
    "path": "/stack",
    "routes": Array [
      Object {
        "component": [Function],
        "exact": true,
        "path": "/stack/charts",
      },
      Object {
        "component": [Function],
        "exact": true,
        "path": "/stack/:repo/:chart",
      },
    ],
  },
  Object {
    "component": undefined,
    "exact": true,
    "path": "/workspace/:id",
  },
]
`;

exports[`Workspace Page checking generatedRoutes 2`] = `
<Switch>
  <Route
    exact={true}
    path="/create-workspace"
    render={[Function]}
    strict={undefined}
  />
  <Route
    exact={true}
    path="/workspace/:id/details"
    render={[Function]}
    strict={undefined}
  />
  <Route
    exact={true}
    path="/workspace/:id/editor"
    render={[Function]}
    strict={undefined}
  />
  <Route
    exact={false}
    path="/stack"
    render={[Function]}
    strict={undefined}
  />
  <Route
    exact={true}
    path="/workspace/:id"
    render={[Function]}
    strict={undefined}
  />
</Switch>
`;

exports[`Workspace Page testing the pageStore 1`] = `
Array [
  Object {
    "/stack/charts": Object {
      "exact": true,
      "key": "stack-charts",
      "name": "Stack",
      "position": "MIDDLE",
    },
  },
]
`;

exports[`Workspace Page testing the pageStore 2`] = `
Array [
  Object {
    "/create-workspace": Object {
      "component": [Function],
      "exact": true,
      "key": "create-workspace",
      "name": "Create Workspace",
      "path": "/create-workspace",
      "position": "MIDDLE",
    },
  },
  Object {
    "/workspace/:id/details": Object {
      "component": [Function],
      "exact": true,
      "key": "workspace-details",
      "name": "Workspace Details",
      "path": "/workspace/:id/details",
      "position": "MIDDLE",
    },
  },
  Object {
    "/workspace/:id/editor": Object {
      "component": [Function],
      "exact": true,
      "key": "workspace-editor",
      "name": "Workspace Editor",
      "path": "/workspace/:id/editor",
      "position": "MIDDLE",
    },
  },
  Object {
    "/stack": Object {
      "component": [Function],
      "exact": false,
      "key": "stack",
      "name": "Stack",
      "path": "/stack",
      "position": "MIDDLE",
    },
  },
  Object {
    "/stack/charts": Object {
      "component": [Function],
      "exact": true,
      "key": "stack-charts",
      "name": "Stack",
      "path": "/stack/charts",
      "position": "MIDDLE",
    },
  },
  Object {
    "/stack/:repo/:chart": Object {
      "component": [Function],
      "exact": true,
      "key": "stack-repo-chart",
      "name": "Stack Chart",
      "path": "/stack/:repo/:chart",
      "position": "MIDDLE",
    },
  },
  Object {
    "/workspace/:id": Object {
      "exact": true,
      "key": "workspace",
      "name": "Workspace",
      "path": "/workspace/:id",
      "position": "MIDDLE",
    },
  },
]
`;
